grass/process.h

```diff
@@ -21,7 +20,7 @@ struct process{
     int pid;
     enum proc_status status;
     uint mepc, saved_register[SAVED_REGISTER_NUM];
-    int receiver_pid; /* used when waiting to send a message */
+    int is_recv;
```

Comment:
Instead of `is_recv`, I think we need something like
```c
enum {
    PENDING_SEND,
    PENDING_RECV,
    PENDING_KEYBOARD_INPUT
} pending_status;
```